tests/Functional/src/com/microsoft/aad/adal/test/StorageHelperTests.java

```diff
@@ -54,18 +57,12 @@ protected void setUp() throws Exception {
         super.setUp();
 
         Log.d(TAG, "setup key at settings");
-        if (AuthenticationSettings.INSTANCE.getSecretKeyData() == null) {
-            // use same key for tests
-            SecretKeyFactory keyFactory = SecretKeyFactory
-                    .getInstance("PBEWithSHA256And256BitAES-CBC-BC");
-            SecretKey tempkey = keyFactory.generateSecret(new PBEKeySpec("test".toCharArray(),
-                    "abcdedfdfd".getBytes("UTF-8"), 100, 256));
-            SecretKey secretKey = new SecretKeySpec(tempkey.getEncoded(), "AES");
-            AuthenticationSettings.INSTANCE.setSecretKey(secretKey.getEncoded());
+        if (AuthenticationSettings.INSTANCE.getSecretKeyData() == null && Build.VERSION.SDK_INT < 18) {
```

Comment:
I think we should actually have 2 different set of tests: 1) if secretkey is null 2) when it's set 
Because I don' think that anyone will launch tests on older versions.